src/vector/backends/numpy.py

```diff
@@ -297,6 +298,14 @@ def _shape_of(result: typing.Tuple[FloatArray, ...]) -> typing.Tuple[int, ...]:
     return tuple(shape)
 
 
+def _is_type_safe(
+    array: typing.Any,
+) -> bool:
+    return issubclass(
+        array.dtype.type, (numpy.integer, numpy.floating)
+    ) and not issubclass(array.dtype.type, numpy.timedelta64)
```

Comment:
`bool` values are automatically converted to 1s and 0s be `numpy` and `numpy` vectors.